xbmc/input/InputManager.cpp

```diff
@@ -480,7 +480,11 @@ bool CInputManager::OnEvent(XBMC_Event& newEvent)
   case XBMC_KEYUP:
     m_Keyboard.ProcessKeyUp();
     if (m_LastKey.GetButtonCode() != KEY_INVALID && !(m_LastKey.GetButtonCode() & CKey::MODIFIER_LONG))
-      OnKey(m_LastKey);
+    {
+      CKey key = m_LastKey;
+      m_LastKey.Reset();  // OnKey is reentrant; need to do this before entering
+      OnKey(key);
+    }
     m_LastKey.Reset();
```

Comment:
@xhaggi Yep, absolutely.
@da-anda That would always create a stack variable, even when it's not needed (i.e. 99% of the time, all the non-longpress cases)